src/lib.rs

```diff
@@ -291,7 +291,7 @@ impl KernelConfig {
 /// implementations are provided here to get a mountable filesystem that does
 /// nothing.
 #[allow(clippy::too_many_arguments)]
-pub trait Filesystem {
+pub trait Filesystem: Clone {
```

Comment:
Sorry for the random drive-by comment. In my opinion, it would be better to require `Send + Sync` instead of clone, and change the trait methods to take non-mutable `self`. Then it's up to the user how they handle mutability, if they need it. Cloning + mutability is a massive footgun.

Reply:
Thanks for the feedback!

I need to figure out exactly what you're referring to and understand it, but that does make sense and does seem better. Something fun to do this weekend!

Reply:
if a type is `Send`, it can be safely moved between threads.

if a type is `Sync`, it can be safely shared between threads.
see https://doc.rust-lang.org/nomicon/send-and-sync.html?highlight=Send#send-and-sync